chrome/content/zotero/elements/contextPane.js

```diff
@@ -218,15 +215,7 @@
 			}
 
 			let currentNoteContext = this._getCurrentNotesContext();
-			let tabNotesDeck = currentNoteContext.querySelector('.zotero-context-pane-tab-notes-deck');
-			let selectedIndex = Array.from(tabNotesDeck.children).findIndex(x => x.getAttribute('data-tab-id') == reader.tabID);
-			if (selectedIndex != -1) {
-				tabNotesDeck.setAttribute('selectedIndex', selectedIndex);
-				currentNoteContext.mode = "childNote";
-			}
-			else {
-				currentNoteContext._restoreViewType();
-			}
+			currentNoteContext.switchToTab(reader.tabID);
```

Comment:
1. Clean up the `_handleTabSelect` and move some dirty code to `NotesContext#switchToTab`